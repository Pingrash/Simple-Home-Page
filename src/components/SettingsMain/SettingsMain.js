import React from 'react';
import PropTypes from 'prop-types';

import Paper from '@material-ui/core/Paper';
import Divider from '@material-ui/core/Divider';
import { makeStyles } from '@material-ui/styles';

import LinkSettings from '../LinkSettings/LinkSettings';
import UserSettings from '../UserSettings/UserSettings';
import SettingsButton from '../SettingsButton/SettingsButton';
import ResetSettings from '../ResetSettings/ResetSettings';

const useStyles = makeStyles(theme => ({
  settingsContainer: {
    width: '70%',
    minWidth: 'fit-content',
    display: 'flex',
    flexDirection: 'column',
    flex: 1,
    padding: '2rem 3rem',
    alignSelf: 'center',
    margin: '1rem 0'
  },
  divider: {
    margin: '2rem 0'
  }
}));

const SettingsMain = ({
  links,
  addLink,
  deleteLink,
  shiftLinkCard,
  toggleSettings,
  user,
  updateUser,
  resetLinks
}) => {
  const classes = useStyles();

  return (
    <Paper elevation={4} className={classes.settingsContainer}>
      <SettingsButton
        toggleSettings={toggleSettings}
        data-test='toggle-button'
      />
      <LinkSettings
        data-test='link-settings'
        links={links}
        addLink={addLink}
        deleteLink={deleteLink}
        shiftLinkCard={shiftLinkCard}
      />
      <Divider variant='middle' className={classes.divider} />
      <UserSettings
        user={user}
        updateUser={updateUser}
        data-test='user-settings'
      />
      <Divider variant='middle' className={classes.divider} />
      <ResetSettings
        updateUser={updateUser}
        resetLinks={resetLinks}
        data-test='reset-settings'
      />
    </Paper>
  );
};

SettingsMain.propTypes = {
  links: PropTypes.arrayOf(PropTypes.object).isRequired,
  addLink: PropTypes.func.isRequired,
  deleteLink: PropTypes.func.isRequired,
  shiftLinkCard: PropTypes.func.isRequired,
  toggleSettings: PropTypes.func.isRequired,
  user: PropTypes.string.isRequired,
  updateUser: PropTypes.func.isRequired,
  resetLinks: PropTypes.func.isRequired
};

export default SettingsMain;
