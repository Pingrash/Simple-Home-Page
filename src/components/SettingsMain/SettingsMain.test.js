import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {
  findByDataTest,
  checkProps,
  testFunc
} from '../../../Utilities';
import SettingsMain from './SettingsMain';

const props = {
  links: [{ link: 'link', favicon: 'favicon', name: 'name' }],
  addLink: testFunc,
  deleteLink: testFunc,
  shiftLinkCard: testFunc,
  toggleSettings: testFunc,
  user: 'username',
  updateUser: testFunc,
  resetLinks: testFunc
};

const setup = () => {
  const component = shallow(<SettingsMain {...props} />);
  return component;
};

describe('Snapshot', () => {
  it('Matches snapshot', () => {
    const tree = renderer
      .create(<SettingsMain {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SettingsMain renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render a toggle button', () => {
    const button = findByDataTest(wrapper, 'toggle-button');
    expect(button.length).toBe(1);
  });

  it('Should render LinkSettings', () => {
    const component = findByDataTest(wrapper, 'link-settings');
    expect(component.length).toBe(1);
  });

  it('Should render UserSettings', () => {
    const component = findByDataTest(wrapper, 'user-settings');
    expect(component.length).toBe(1);
  });

  it('Should render ResetSettings', () => {
    const component = findByDataTest(wrapper, 'reset-settings');
    expect(component.length).toBe(1);
  });
});

describe('SettingsMain should have correct props', () => {
  it('Should not return any props errors', () => {
    const propsError = checkProps(SettingsMain, props);
    expect(propsError).toBeUndefined();
  });
});
