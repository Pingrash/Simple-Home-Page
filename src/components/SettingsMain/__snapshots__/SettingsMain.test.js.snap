// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Matches snapshot 1`] = `
<div
  className="MuiPaper-root makeStyles-settingsContainer-1 MuiPaper-elevation4 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-31"
    data-test="button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Settings
    </span>
  </button>
  <form
    className="makeStyles-addForm-67"
    data-test="entry-form"
    onSubmit={[Function]}
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      Add Link
    </h4>
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="url"
      >
        Website URL:
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        data-test="form-field"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          autoFocus={true}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="url"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="https://www.google.com"
          required={false}
          type="url"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="name"
      >
        Link Name:
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        data-test="form-field"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="name"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitButton-65 MuiButton-containedPrimary MuiButton-fullWidth"
        data-test="submit-button"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </form>
  <div
    className="makeStyles-cardsContainer-64"
  >
    <div
      className="makeStyles-mainContainer-159"
    >
      <div
        className="makeStyles-arrowContainer-160"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-arrow-161"
          data-test="shift-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-arrow-161"
          data-test="shift-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="MuiPaper-root makeStyles-cardContainer-154 MuiPaper-elevation1 MuiPaper-rounded"
        data-test="container"
      >
        <img
          alt="favicon"
          className="makeStyles-cardFavicon-156"
          data-test="favicon"
          src="favicon"
        />
        <span
          className="MuiTypography-root makeStyles-cardTitle-157 MuiTypography-h6"
          data-test="title"
        >
          name
        </span>
        <span
          className="MuiTypography-root makeStyles-cardUrl-158 MuiTypography-body2"
          data-test="url"
        >
          link
        </span>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text card-button MuiButton-textSecondary"
          data-test="delete-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Delete
          </span>
        </button>
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root makeStyles-divider-2 MuiDivider-middle"
  />
  <form
    className="makeStyles-form-177"
    data-test="form"
    onSubmit={[Function]}
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      Update User
    </h4>
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="username-input"
      >
        New Username:
         
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-formSpacing-178 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="username-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="username"
          required={false}
          type="text"
        />
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formSpacing-178 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </form>
  <hr
    className="MuiDivider-root makeStyles-divider-2 MuiDivider-middle"
  />
  <h4
    className="MuiTypography-root MuiTypography-h4"
  >
    Reset To Default
  </h4>
  <span
    className="MuiTypography-root makeStyles-spacing-180 MuiTypography-body1 MuiTypography-colorError"
    data-test="warning"
  >
    WARNING: Pressing this button will reset everything to default!
  </span>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-spacing-180 MuiButton-outlinedSecondary"
    data-test="reset-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Delete User
    </span>
  </button>
</div>
`;
